src/utils/filter.py

```
@@ -0,0 +1,536 @@
+    Returns:
+        Combined condition object or None if no conditions
+    """
+    if not conditions:
+        return None
+    elif len(conditions) == 1:
+        return conditions[0]
+    else:
+        return and_(*conditions)
+
+
+def _build_comparison_conditions(
```

Ruff (0.11.9)

`_build_comparison_conditions` is too complex (27 > 10)

(C901)